tracehandler/tracehandler.go

```diff
@@ -77,29 +76,45 @@ type traceHandler struct {
 }
 
 // newTraceHandler creates a new traceHandler
-func newTraceHandler(rep reporter.TraceReporter, traceProcessor TraceProcessor,
-	intervals Times, cacheSize uint32) (*traceHandler, error) {
-	bpfTraceCache, err := lru.New[host.TraceHash, libpf.TraceHash](
+func newTraceHandler(ctx context.Context, rep reporter.TraceReporter,
+	traceProcessor TraceProcessor, intervals Times, cacheSize uint32) (*traceHandler, error) {
+	traceCache, err := lru.NewSynced[host.TraceHash, *libpf.Trace](
 		cacheSize, func(k host.TraceHash) uint32 { return uint32(k) })
 	if err != nil {
 		return nil, err
 	}
+	// Do not hold elements indefinitely in the cache.
+	traceCache.SetLifetime(traceCacheLifetime)
 
-	umTraceCache, err := lru.New[libpf.TraceHash, libpf.Void](
-		cacheSize, libpf.TraceHash.Hash32)
-	if err != nil {
-		return nil, err
-	}
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	go func() {
+		wg.Done()
+		jitter := 0.2
+		ticker := time.NewTicker(libpf.AddJitter(traceCacheLifetime, jitter))
```

Comment:
We probably shouldn't be adding jitter here (which can increase/decrease frequency of purges by 20%) as purging an LRU is linear to its size, and its size (minimum: 65536) depends on number of cores. I'd prefer deterministic-regular behavior in purging as that is easier to look at metrics and figure out if it is a CPU sink and we need to further optimize.

Reply:
Dropped the jitter with https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/405/commits/c3c73270d7b32b37d83c48a2d3935848f4c4eaed. No strong opinion here.

Reply:
For follow-up investigations:
I wonder if the cache size calculation needs further refinement. A size of 65536 is used for numCores < 115 (at 5s monitoring interval and 19Hz sampling frequency), with more cores the size jumps to 128k. The cache size calculation should also consider the lifetime of items.
We could start with a metric for the number of cached traces. With enough data, we can come up with a better formula for the cache size.